sender.go

```diff
@@ -97,6 +98,24 @@ func (s *Sender) Send(to *Address, method string, args ...EtherType) (Hash, erro
 	return s.Call(&opts)
 }
 
+// Cancel a transaction with the given hash.
+func (s *Sender) Cancel(h *Hash) (Hash, error) {
+	tx, err := s.GetTransaction(h)
+	if err != nil {
+		return Hash{}, err
+	} else if tx.TxIndex != nil {
+		return Hash{}, errors.New("seth: cannot cancel")
```

Comment:
I think this error should be a top-level error so that callers can easily recognize when cancellation has failed. (My rationale is that failing to cancel a transaction means, at a high level, that the overall operation was 'successful,' since we'd likely only be cancelling transactions in order to re-post them.

Reply:
I'm considering removing this check since there might be a good reason to try to post a transaction even though a transaction with the same nonce might already have been mined. Instead, I'll add a method to check the status of a transaction easily and allow callers to do that error handling themselves.

Reply:
Nevermind, I just made it a global error.